### PR TITLE
Model 10 changes

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description
- *Category*: bugfix/feature
- *JIRA issue*: [MIC-4657](https://jira.ihme.washington.edu/browse/MIC-4657)
- *Research reference*: [wasting transitions](https://vivarium-research.readthedocs.io/en/latest/models/risk_exposures/child_growth_failure/wasting/wasting_state_exposure/index.html#vivarium-modeling-strategy) and [birth prevalence equations](https://vivarium-research.readthedocs.io/en/latest/models/risk_exposures/child_growth_failure/wasting/wasting_state_exposure/index.html#id5)

### Changes and notes
Update birth prevalence equations and use dynamic transition wasting model for all ages.
Some linting changes.

### Verification and Testing
Remade artifact, checked that prevalences summed up to 1, and ran sim for a few time steps.

### CODE REVIEW

File: src/vivarium_gates_nutrition_optimization_child/components/observers.py [hunk 1]
```diff
@@ -33,20 +33,20 @@ class ResultsStratifier(ResultsStratifier_):
...
-#            "age_end": [0.5, 1.5, 5],
-#            "age_group_name": [
-#                "0_to_6_months",
-#                "6_to_18_months",
-#                "18_to_59_months",
-#            ],
-#        }
-#
-#        return pd.DataFrame(data_dict)
+    #    def get_age_bins(self, builder: Builder) -> pd.DataFrame:
```
Comment (hussain-jafari) [inline src/vivarium_gates_nutrition_optimization_child/components/observers.py:36]: linting

File: src/vivarium_gates_nutrition_optimization_child/constants/results.py [hunk 2]
```diff
@@ -26,7 +26,9 @@
 
 THROWAWAY_COLUMNS = [f"{state}_event_count" for state in models.STATES]
 
-DEATH_COLUMN_TEMPLATE = "MEASURE_death_due_to_{CAUSE_OF_DEATH}_AGE_GROUP_{AGE_GROUP}_SEX_{SEX}"
+DEATH_COLUMN_TEMPLATE = (
```
Comment (hussain-jafari) [inline src/vivarium_gates_nutrition_optimization_child/constants/results.py:29]: linting

File: src/vivarium_gates_nutrition_optimization_child/components/wasting.py [hunk 3]
```diff
@@ -245,8 +204,13 @@ def setup(self, builder):
...
 
     def on_initialize_simulants(self, pop_data):
+        initial_propensity = self.randomness.get_draw(pop_data.index).rename(
+            f"initial_{self.state_column}_propensity"
+        )
+        self.population_view.update(initial_propensity)
+
         population = self.population_view.subview(
-            ["age", "sex", "static_child_wasting_propensity"]
+            ["age", "sex", "initial_child_wasting_propensity"]
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/wasting.py:213]: How does this work? Doesn't this component create the `initial_child_wasting_propensity` column? This should error when initializing the population, as this column won't exists yet.

File: src/vivarium_gates_nutrition_optimization_child/components/wasting.py [hunk 4]
```diff
@@ -245,8 +204,13 @@ def setup(self, builder):
         )
 
     def on_initialize_simulants(self, pop_data):
+        initial_propensity = self.randomness.get_draw(pop_data.index).rename(
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/wasting.py:207]: I don't see where you use this.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_gates_nutrition_optimization_child/components/wasting.py (hunk 4 above)
Comment (hussain-jafari) [reply src/vivarium_gates_nutrition_optimization_child/components/wasting.py:207]: I update the state table to include these propensities in line 210

File: src/vivarium_gates_nutrition_optimization_child/components/wasting.py [hunk 5]
```diff
@@ -235,13 +231,13 @@ def on_initialize_simulants(self, pop_data):
...
             )
-        self.population_view.update(condition_column)
+        self.population_view.update(pd.concat([condition_column, initial_propensity], axis=1))
 
     @staticmethod
     def assign_initial_status_to_simulants(
         simulants_df, state_names, weights_bins, propensities
     ):
-        simulants = simulants_df[["age", "sex", "initial_child_wasting_propensity"]].copy()
+        simulants = simulants_df[["age", "sex"]].copy()
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/wasting.py:240]: Can you add typing to this method. Also the variable names are a bit confusing to me. This function is actually returning a dataframe, but the method calling it acts like it is only returning a single column. I think only returning the column is more correct, but at least adding typing would make this clear.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**